editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-strategy-helpers.ts

```diff
@@ -0,0 +1,159 @@
+import { MetadataUtils } from '../../../../../core/model/element-metadata-utils'
+import { ElementInstanceMetadataMap } from '../../../../../core/shared/element-template'
+import { CanvasPoint, Size } from '../../../../../core/shared/math-utils'
+import { ElementPath } from '../../../../../core/shared/project-file-types'
+import {
+  getTargetPathsFromInteractionTarget,
+  InteractionCanvasState,
+  InteractionTarget,
+} from '../../canvas-strategy-types'
+import { AllowSmallerParent } from '../../interaction-state'
+import {
+  flowParentAbsoluteOrStatic,
+  getReparentTargetUnified,
+} from './reparent-strategy-parent-lookup'
+import { getDragTargets } from '../shared-move-strategies-helpers'
+
+export type ReparentStrategy = 'REPARENT_AS_ABSOLUTE' | 'REPARENT_AS_STATIC'
+
+export type FindReparentStrategyResult = {
+  strategy: ReparentStrategy
+  isFallback: boolean
+  target: ReparentTarget
+}
+
+export function reparentStrategyForPaste(
```

Comment:
this is the same as master, just with a new name to indicate that this is _only_ used by the paste action